codegen/codegen_test.go

```diff
@@ -53,6 +54,86 @@ func TestCodeGen(t *testing.T) {
 		func(t *testing.T, cg *CodeGen) solver.Request {
 			return Expect(t, llb.Image("alpine"))
 		},
+	}, {
+		"basic scratch",
+		[]string{"default"},
+		`
+		fs default() {
+			scratch
+		}
+		`,
+		func(t *testing.T, cg *CodeGen) solver.Request {
+			return Expect(t, llb.Scratch())
+		},
+	}, {
+		"basic http",
+		[]string{"default"},
+		`
+		fs default() {
+			http "http://my.test.url"
+		}
+		`,
+		func(t *testing.T, cg *CodeGen) solver.Request {
+			return Expect(t, llb.HTTP("http://my.test.url"))
+		},
+	}, {
+		"http with options",
+		[]string{"default"},
+		`
+		fs default() {
+			http "http://my.test.url" with option {
+				checksum "123"
+				chmod 777
+				filename "myTest.out"
+			}
+		}
+		`,
+		func(t *testing.T, cg *CodeGen) solver.Request {
+			return Expect(t, llb.HTTP(
+				"http://my.test.url",
+				llb.Checksum("123"),
+				llb.Chmod(os.FileMode(777)),
+				llb.Filename("myTest.out")))
+		},
+	}, {
+		"basic git",
+		[]string{"default"},
+		`
+		fs default() {
+			git "https://github.com/openllb/hlb.git" "master"
+		}
+		`,
+		func(t *testing.T, cg *CodeGen) solver.Request {
+			return Expect(t, llb.Git("https://github.com/openllb/hlb.git", "master"))
+		},
+	}, {
+		"git with options",
+		[]string{"default"},
+		`
+		fs default() {
+			git "https://github.com/openllb/hlb.git" "master" with option {
+				keepGitDir
+			}
+		}
+		`,
+		func(t *testing.T, cg *CodeGen) solver.Request {
+			return Expect(t, llb.Git(
+				"https://github.com/openllb/hlb.git",
+				"master",
+				llb.KeepGitDir()))
+		},
+	}, {
+		"basic env",
+		[]string{"default"},
+		`
+		fs default() {
+			scratch
+			env "TEST_VAR" "test value"
+		}
+		`,
+		func(t *testing.T, cg *CodeGen) solver.Request {
+			return Expect(t, llb.Scratch().AddEnv("TEST_VAR", "test value"))
```

Comment:
@hinshun Heads up that this is actually a no-op. The environment is taken into consideration when displaying the Tree.

Reply:
Yeah, that's fine. I'm okay with this.